### PR TITLE
[rotel] Refactor to separate comm handling from protocol handling

### PR DESCRIPTION
**Pull request by lolodomo**
Each protocol is now handled in a separate class.

Allow to reduce the size of class RotelConnector

Signed-off-by: Laurent Garnier <lg.hc@free.fr>

### CODE REVIEW

**Comment by lolodomo** [comment]
A note for reviewers: this is only a technical refactoring with code moved from class RotelConnector to other classes. As there is no real new (functional) code, I really don't think there is a need for a detailed review.

**Comment by lolodomo** [comment]
Please first merge #12524 ; I will then resolve the conflicts.

**Comment by lolodomo** [comment]
@fwolter @jlaur : any chance to merge this PR ? As previously mentioned, this is mainly code moved from one class to (new) other classes. I don't think there is the need of a detailed review.
I am running this version without any problem.

File: bundles/org.openhab.binding.rotel/src/main/java/org/openhab/binding/rotel/internal/protocol/hex/RotelHexProtocolHandler.java [hunk 1]
```diff
@@ -0,0 +1,778 @@
...
+                            RotelFlagsMapping.isBitFlagOn(flags, i, 2), RotelFlagsMapping.isBitFlagOn(flags, i, 1),
+                            RotelFlagsMapping.isBitFlagOn(flags, i, 0));
+                } catch (RotelException e1) {
+                }
+            }
+        }
+        try {
+            dispatchKeyValue(KEY_POWER_ZONE2, model.isZone2On(flags) ? POWER_ON : STANDBY);
+        } catch (RotelException e1) {
+        }
```
Comment (fwolter) [inline bundles/org.openhab.binding.rotel/src/main/java/org/openhab/binding/rotel/internal/protocol/hex/RotelHexProtocolHandler.java:341]: What's the reason all these catch blocks are empty? If this is intentional, you could add a comment
```suggestion
        try {
            dispatchKeyValue(KEY_POWER_ZONE2, model.isZone2On(flags) ? POWER_ON : STANDBY);
        } catch (RotelException e1) {
            // nothing
        }
```

**Review by fwolter — CHANGES_REQUESTED** [review]

File: bundles/org.openhab.binding.rotel/src/main/java/org/openhab/binding/rotel/internal/protocol/hex/RotelHexProtocolHandler.java (hunk 1 above)
Comment (lolodomo) [reply bundles/org.openhab.binding.rotel/src/main/java/org/openhab/binding/rotel/internal/protocol/hex/RotelHexProtocolHandler.java]: Thank you @fwolter for your review.
I checked all RotelException catch and added a comment when nothing was done.

**Review by fwolter — APPROVED** [review]

**Comment by lolodomo** [comment]
Big thank you @fwolter

**Pull request merged**